client/src/partials/reports/balance_mensuelle/balance_mensuelle.js

```diff
@@ -90,29 +92,98 @@ function ReportBalanceMensuelleController ($translate, $window, validate, util,
     });
   }
 
-  function getAccountBalance() {
+  // actually submits the form
+  function submit(invalid) {
+    var url;
+
+    if (invalid) { return; }
+
     vm.state = 'generate';
-    if (session.classe && session.periode) {
+    vm.loading = true;
 
-      var params = {
-        project : session.project,
-        classe  : session.classe.number,
-        periode : util.sqlDate(session.periode)
-      };
+    // empty data caches
+    vm.accounts = {};
+    vm.totals = {};
 
-      var rootReport = 'balance_mensuelle';
+    url = '/reports/balance_mensuelle?' +
+        'enterpriseId=' + Session.enterprise.id +
+        '&classe=' + session.classe.number +
+        '&date=' + Dates.util.str(session.periode);
 
-      if (session.classe.number === '*') {
-        rootReport = 'balance_mensuelle_all';
-      }
+    // get the data
+    $http.get(url)
+    .then(function (response) {
+      var accounts,
+          totals,
+          data = response.data;
+
+      // make the accounts object
+      accounts = data.beginning.reduce(function (accounts, row) {
+        var account,
+            id = row.account_number;
+
+        if (!accounts[id]) { accounts[id] = {}; }
+        account = accounts[id];
+        account.beginDebit = row.debit;
+        account.beginCredit = row.credit;
+        account.account_txt = row.account_txt;
+        return accounts;
+      }, {});
 
-      dependencies.balance_mensuelle = { query : 'reports/' + rootReport + '/?' + JSON.stringify(params) };
-      validate.process(dependencies, ['balance_mensuelle'])
-      .then(initialize)
-      .then(calculTotaux);
-    } else {
-      messenger.info($translate.instant('BALANCE_MENSUELLE.INFO_FILL_INPUT'), true);
-    }
+      // add the middle references
+      data.middle.forEach(function (row) {
+        var account,
+            id = row.account_number;
+
+        if (!accounts[id]) { accounts[id] = {}; }
+        account = accounts[id];
+        account.middleDebit = row.debit;
+        account.middleCredit = row.credit;
+        account.account_txt = row.account_txt;
+      });
+
+      // add the final balances
+      data.end.forEach(function (row) {
+        var account,
+            id = row.account_number;
+
+        if (!accounts[id]) { accounts[id] = {}; }
+        account = accounts[id];
+        account.endDebit = row.debit;
+        account.endCredit = row.credit;
+        account.account_txt = row.account_txt;
+      });
+
+      // calculate totals
+      totals = Object.keys(accounts).reduce(function (totals, key) {
+        var account = accounts[key];
+        totals.beginDebit += (account.beginDebit || 0);
```

Comment:
There is a lot repetitive code here, this isn't necessarily a bad thing because it is very clear what the code does. 

This could be done by iterating through the objects keys and incrementing (as the two objects always share the same key pairs). It could also be done by modelling the object as a class and writing an increment function that accepts another class (if this is used in enough places to warrant this). 

Defining the initial object passed to the filter would help provide a semantic name for the second parameter, as well as provide a clear origin of the object keys.